src/gluonts/mx/model/n_beats/_ensemble.py

```diff
@@ -439,14 +439,33 @@ def from_hyperparameters(
             raise GluonTSHyperparametersError from e
 
     def train(
-        self, training_data: Dataset, validation_data: Optional[Dataset] = None
+        self,
+        training_data: Dataset,
+        validation_data: Optional[Dataset] = None,
+        model_init: Optional[Predictor] = None,
     ) -> NBEATSEnsemblePredictor:
         predictors = []
 
-        for index, estimator in enumerate(self.estimators):
-            logging.info(
-                f"Training estimator {index + 1}/{len(self.estimators)}."
-            )
-            predictors.append(estimator.train(training_data, validation_data))
+        if model_init is None:
+            for index, estimator in enumerate(self.estimators):
+                logging.info(
+                    f"Training estimator {index + 1}/{len(self.estimators)}."
+                )
+                predictors.append(
+                    estimator.train(training_data, validation_data)
+                )
+        else:
+            assert isinstance(model_init, NBEATSEnsemblePredictor)
```

Comment:
I think this is weird type-wise. Each `Estimator` class has exactly one companion predictor-class. I guess we would want something like Rust's [Associated types](https://doc.rust-lang.org/rust-by-example/generics/assoc_items/types.html#associated-types)

Reply:
This would just be a class attribute in this context, right?